tests/test_admin.py

```diff
@@ -457,12 +460,42 @@ def test_filer_upload_permissions_error(self, extra_headers={}):
         self.assertContains(response, NO_PERMISSIONS_FOR_FOLDER)
         self.assertEqual(Image.objects.count(), 0)
 
+    def test_filer_ajax_upload_without_permissions_error(self, extra_headers={}):
+        """User without add_file permission cannot upload"""
+        self.client.logout()
+        staff_user = User.objects.create_user(
+            username='joe_new', password='x', email='joe@mata.com')
+        staff_user.is_staff = True
+        staff_user.save()
+        self.client.login(username='joe_new', password='x')
+        self.assertEqual(Image.objects.count(), 0)
+        folder = Folder.objects.create(name='foo')
+        file_obj = django.core.files.File(open(self.filename, 'rb'))
```

Comment:
This is a valid critique of the code, can we use `with` to open the file here and do things under it's indent.

Reply:
Luckily only in the tests. But I agree that's bad coding style.